src/components/TimeZonePicker.vue

```diff
@@ -0,0 +1,111 @@
+<!-- ADD_LICENSE_HEADER -->
+<template>
+    <div class="input-group">
+        <div class="input-group-prepend">
+            <span class="input-group-text">Zone</span>
+        </div>
+        <input
+            :key="componentKey"
+            v-bind:value="timezoneString"
+            v-on:input="validateTimeZone"
+            v-on:change="submitChange"
+            placeholder="[+-]hh:mm"
+            class="form-control" />
+    </div>
+</template>
+
+<script>
+
+export default {
+	name: 'timezonepicker',
+	components: {
+    },
+    model: {
+        prop: 'value',
+        event: 'input'
+    },
+    props: {
+        format: String,
+        value: String,
+    },
+	data: function () {
+        return {
+            internal_value: null,
+            initial_value: null,
+            isInitializing: true,
+            componentKey: 0,
+            timezoneRegexp: /^[+-]?((0[0-9]|1\d|2[0-3]|[0-9])|((0[0-9]|1\d|2[0-3]|[0-9]):){1})$/
+        }
+	},
+	computed: {
+        timezoneString: {
+			get: function() {
+                return this.internal_value
+			},
+			set: function(new_value) {
+                if (this.isInitializing) { return }
+				if (this.timezoneRegexp.test(new_value)) {
+                    this.internal_value = new_value
+				}
+			}
+        },
+	},
+	methods: {
+        submitChange: function(event) {
```

Comment:
Style thing: In other components changes are emited as internal value changes (watch), in this it is on change. I (currently, my style is in a bit flux) like the idea that watch is used to update parent. It moves the update logic dependency from template to the js part that in my opinion makes it easier to follow the data flow. In theory, props --> template, template event --> internal state, internal state --> update parent (or store). Hopefully this doesn't cause confusion, I'm not rely sure I delivered the reasoning well. And in any case this is not a change request but rather a note on style.

Reply:
I can change this in the next PR when I add the helpers.

Reply:
might be improved already